src/views/CardStack.js

```diff
@@ -207,11 +224,16 @@ class CardStack extends React.Component<DefaultProps, Props, void> {
           if (header && header.left) {
             return header.left;
           }
-          const { renderLeftComponent } = this.props.headerComponent.defaultProps || {};
-          if (typeof renderLeftComponent === 'function') {
-            return renderLeftComponent(props);
+          if (props.scene.index === 0 || !props.onNavigateBack) {
+            return null;
           }
-          return null;
+          const prevSceneTitle = this._getHeaderTitleForScene(props.scenes[props.index - 1]);
+          return (
+            <HeaderBackButton
```

Comment:
Decided to fall back to original approach as it was just easier to define custom back button that way (specify custom title). Otherwise I would've had to change either the signature of `renderLeftComponent` inside Header or make Header itself aware of a previous scene which I wanted to avoid.